syntax = "proto3";

message GMetaMsg{
    string msg_name = 1;
    int32 data_size = 2;
}

message GProxyMsg{
    enum Cmd
    {
        CREATE = 0;
        DELETE = 1;
    }
    Cmd cmd     = 1;
    string uuid = 2;
}

message GTransformMsg{
    enum Cmd
    {
        CREATE = 0;
        UPDATE = 1;
        DELETE = 2;
    }
    Cmd cmd      = 1;
    string suuid = 2;
    string cuuid = 3;
    float posx   = 4;
    float posy   = 5;
    float posz   = 6;
    float rx     = 7;
    float ry     = 8;
    float rz     = 9;
    float sx     = 10;
    float sy     = 11;
    float sz     = 12;
}
